metrics/time/launch_times.sh

```diff
@@ -38,6 +38,34 @@ REQUIRED_CMDS=("bc" "awk")
 # Note the no-0-padding - 0 padding the results breaks bc in some cases
 DATECMD="date -u +%-s:%-N"
 
+# This the minimum entropy level produced
+# by haveged is 1000 see https://wiki.archlinux.org/index.php/Haveged
+# Less than 1000 could potentially slow down cryptographic
+# applications see https://www.suse.com/support/kb/doc/?id=7011351
+entropy_level="1000"
+
+check_entropy_package() {
+	entropy_package="haveged"
+	# Check that entropy package is installed
+	check_package=$(which ${entropy_package} | wc -l)
+	if [ ${check_package} -eq 0 ]; then
+		die "No entropy package installed"
+	fi
+	# Check that service is running
+	sudo service ${entropy_package} status | grep "running"
+}
+
+check_entropy_level() {
+	retries="10"
+	for i in $(seq 1 "$retries"); do
+		if [ $(cat /proc/sys/kernel/random/entropy_avail) -ge ${entropy_level} ]; then
+			break;
+		else
+			die "Low entropy level"
```

Comment:
won't this just die on the first time around the loop - should the die be after the `for` loop?
But, yay, for the sleepy retry :-)